### ISSUE TITLE
Add configuration for 

### ISSUE THREAD

**Issue by westonganger** [salsify/goldiloader#55]
Is there the ability to set goldiloader to set the whole application to `auto_include(false)` by default? I only want to use goldiloader for certain controller actions. I think I need something like:

```ruby
Goldiloader.config do |config|
 config.auto_include = false
end
```

Then when I want to use it I can do the following

```ruby
@posts = Post.auto_include # preferable
# or
@posts = Post.auto_include(true)
```

**Comment by jturkel** [comment]
We don't currently have this configuration option. Can you elaborate on why you only want to use goldiloader for certain controller actions (or is it certain models)? Also note the `auto_include` query scope method does not apply recursively e.g. `Blog.auto_include` will enable automatic eager loading of the blog's `posts` associations but it will not enable automatic eager loading of the post's associations.

**Comment by westonganger** [comment]
Yeah it was for nested includes. That kills the main purpose for this. However it may be nice to implement this so its not an all or nothing approach.

**Comment by jturkel** [comment]
One possibility would be to implement a block form of `auto_include` that would apply for the scope of the block e.g.

```
Blog.auto_include do
  Blog.all.find_each do |blog|
    posts = blog.posts
    ...
  end
end
```

**Comment by westonganger** [comment]
Thats another interesting idea which could make this library pretty powerful. I think both of these ideas should be added.

**Comment by jturkel** [comment]
I'm totally open to implementing this enhancement but can you elaborate on why you don't want to enable goldiloader by default? What are the scenarios where it's not working for you?

**Comment by westonganger** [comment]
Well it was to do with the nested includes, that argument is now void. Its just a lot of magic, I could see some people might like it be more explicit and opt-in only when they want, they should have that option.

**Comment by sobrinho** [comment]
@jturkel I can make a PR for this one.

Just wondering if that makes sense to implement! :)

**Comment by jturkel** [comment]
I'd be open to a change that adds a global configuration for enabling/disabling golidloader (as long as the default is enabled) + a block form of `auto_include(enabled = true)` as long as it interacts cleaning with the `auto_include` query scope introduced in #52.
